### PR TITLE
Bug 1948311: DelegatingAuthenticationOptions TokenReview request timeout

### PR DESCRIPTION
**Pull request by ingvagabund**
Reapply https://github.com/openshift/aws-ebs-csi-driver-operator/pull/121 undone by https://github.com/openshift/aws-ebs-csi-driver-operator/pull/116

### CODE REVIEW

File: vendor/k8s.io/apiserver/pkg/server/options/authentication.go [hunk 1]
```diff
@@ -419,7 +420,10 @@ func (s *DelegatingAuthenticationOptions) getClient() (kubernetes.Interface, err
 	// set high qps/burst limits since this will effectively limit API server responsiveness
 	clientConfig.QPS = 200
 	clientConfig.Burst = 400
-	clientConfig.Timeout = s.ClientTimeout
+	// do not set a timeout on the http client, instead use context for cancellation
+	// if multiple timeouts were set, the request will pick the smaller timeout to be applied, leaving other useless.
+	//
+	// see https://github.com/golang/go/blob/a937729c2c2f6950a32bc5cd0f5b88700882f078/src/net/http/client.go#L364
```
Comment (p0lyn0mial) [inline vendor/k8s.io/apiserver/pkg/server/options/authentication.go:426]: this is what we need.

File: vendor/k8s.io/apiserver/plugin/pkg/authenticator/token/webhook/webhook.go [hunk 2]
```diff
@@ -105,7 +106,17 @@ func (w *WebhookTokenAuthenticator) AuthenticateToken(ctx context.Context, token
...
 		result *authenticationv1.TokenReview
 		auds   authenticator.Audiences
+		cancel context.CancelFunc
 	)
+
+	// set a hard timeout if it was defined
+	// if the child has a shorter deadline then it will expire first,
+	// otherwise if the parent has a shorter deadline then the parent will expire and it will be propagate to the child
+	if w.requestTimeout > 0 {
+		ctx, cancel = context.WithTimeout(ctx, w.requestTimeout)
```
Comment (p0lyn0mial) [inline vendor/k8s.io/apiserver/plugin/pkg/authenticator/token/webhook/webhook.go:116]: and this

**Comment by jsafrane** [comment]
/approve
/override ci/prow/e2e-aws-csi
snapshots are broken, working on it

**Pull request merged**